auth/store.go

```diff
@@ -34,10 +35,11 @@ var (
 
 	plog = capnslog.NewPackageLogger("github.com/coreos/etcd", "auth")
 
-	ErrUserAlreadyExist = errors.New("auth: user already exists")
-	ErrUserNotFound     = errors.New("auth: user not found")
-	ErrRoleAlreadyExist = errors.New("auth: role already exists")
-	ErrRoleNotFound     = errors.New("auth: role not found")
+	ErrUserAlreadyExist   = errors.New("auth: user already exists")
+	ErrUserNotFound       = errors.New("auth: user not found")
+	ErrUserAlreadyGranted = errors.New("auth: a user is already granted a role")
```

Comment:
OK, I'll change the checking logic for simply ignoring the request with info logging.